[Expanded Weights] fix conv3d

samdow opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80944
* __->__ #80943
* #80895
* #80892

Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

Commit:
[Expanded Weights] fix conv3d

[ghstack-poisoned]

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

[ghstack-poisoned]

zou3519 commented on test/test_expanded_weights.py:
```diff
@@ -480,10 +480,17 @@ def test_context_manager_multiple_inputs(self, test_case, device):
 # These currently use the legacy nn tests
 supported_modules = ['Linear', 'Conv1d', 'Conv2d', 'Conv3d', 'Embedding', 'LayerNorm', 'GroupNorm', 'InstanceNorm']
-supported_tests = [t for t in module_tests + new_module_tests if 'module_name' in t and t['module_name'] in supported_modules]
+supported_tests = [t for t in module_tests + new_module_tests if filter_supported_tests(t)]
```
(no action required) As this PR (and the last one) exposed, we're not really sure of how much coverage we actually have on expanded weights due to the legacy nn tests and OpInfo not having coverage. We probably want to identify the set of modules ExpandedWeights works over (you probably have this already) and then figure out if we do have the right amount of test coverage

jbschlosser commented on test/test_expanded_weights.py:
```diff
@@ -480,10 +480,17 @@ def test_context_manager_multiple_inputs(self, test_case, device):
+        return True
+    if 'fullname' in t and any([module + "_" in t['fullname'] for module in supported_modules]):
+        return not('Conv' in t['fullname'] and 'pad' in t['fullname'])
+
```
Yes definitely! I'll take this on

jbschlosser commented on test/test_expanded_weights.py:
```diff
@@ -480,10 +480,17 @@ def test_context_manager_multiple_inputs(self, test_case, device):
+
 # TODO: Once all of these use ModuleInfo, replace with ModuleInfo tests
 # These currently use the legacy nn tests
 supported_modules = ['Linear', 'Conv1d', 'Conv2d', 'Conv3d', 'Embedding', 'LayerNorm', 'GroupNorm', 'InstanceNorm']
```
Fair to say that this list should be prioritized for ModuleInfo entries from your perspective?

samdow replied:
Yes! It would be really helpful to avoid these sorts of user/filtering errors to have them in ModuleInfos

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

[ghstack-poisoned]

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

[ghstack-poisoned]

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

[ghstack-poisoned]

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

[ghstack-poisoned]

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

[ghstack-poisoned]

Commit:
Update on "[Expanded Weights] fix conv3d"


Conv3d has an ordering typo in its unfold (it needs a custom unfold since unfold3d doesn't exist in torch). This was caught by Opacus but not by us because dilation, padding, and stride always matched in the test cases. Conv3d has very few test cases since it doesn't have an OpInfo and we were skipping some by the nn module testing filtering. So this updated this filtering to add more of the common nn tests (one of which did fail without the change) 

Closes #80953

[ghstack-poisoned]
